src/AspNetDotEnvConfigurationProvider/AspNetDotEnvConfigurationProvider.cs

```diff
@@ -3,15 +3,95 @@
 using System.Linq;
 using System.Threading.Tasks;
 using Microsoft.Extensions.Configuration;
+using System.IO;
+using System.Collections;
 
 namespace AspNetDotEnvConfigurationProvider
 {
-    // This project can output the Class library as a NuGet Package.
-    // To enable this option, right-click on the project and select the Properties menu item. In the Build tab select "Produce outputs on build".
-    public class AspNetDotEnvConfigurationProvider : ConfigurationProvider
-    {
-        public AspNetDotEnvConfigurationProvider()
-        {
-        }
-    }
+	// This project can output the Class library as a NuGet Package.
+	// To enable this option, right-click on the project and select the Properties menu item. In the Build tab select "Produce outputs on build".
+	public class AspNetDotEnvConfigurationProvider : ConfigurationProvider
+	{
+		public string Path { get; }
```

Comment:
Can these be made private?
Also lets just deal with FileInfo instead of a strings.
```
public AspNetDotEnvConfigurationProvider(FileInfo path, bool optional = false)
 +		{
 +			if (path == null)
 +			{
 +				throw new ArgumentException($"Invalid file path: {path.Filename}", nameof(path));
 +			}

 +			if (path.Exists && !optionsal )
throw new FileNotFoundException($"File not found, path: {file.Filename}", Path);
 +
 +			Optional = optional;
 +			Path = path;
 +		}
```